client/state/jetpack-connect/actions.js

```diff
@@ -197,28 +196,34 @@ export function retryAuth( url, attemptNumber ) {
 }
 
 export function createAccount( userData ) {
-	return dispatch => {
-		dispatch( recordTracksEvent( 'calypso_jpc_create_account', {} ) );
+	return async dispatch => {
+		dispatch( recordTracksEvent( 'calypso_jpc_create_account' ) );
 
-		dispatch( { type: JETPACK_CONNECT_CREATE_ACCOUNT } );
-		wpcom.undocumented().usersNew( userData, ( error, data ) => {
-			if ( error ) {
-				dispatch(
-					recordTracksEvent( 'calypso_jpc_create_account_error', {
-						error: JSON.stringify( error ),
-						error_code: error.code,
-					} )
-				);
-			} else {
-				dispatch( recordTracksEvent( 'calypso_jpc_create_account_success' ) );
-			}
-			dispatch( {
-				type: JETPACK_CONNECT_CREATE_ACCOUNT_RECEIVE,
-				data,
-				error,
-				userData,
-			} );
-		} );
+		try {
+			const data = await wpcom.undocumented().usersNew( userData );
+			const bearerToken = makeParser(
```

Comment:
`makeParser` was added for data-layer, but it's handy for more things like this. I'd like to "promote" it to a framework level util (`client/lib/make-json-schema-parser`).